Dockerfile

```diff
@@ -15,9 +21,8 @@ ENV DISTRIBUTION=$DISTRIBUTION
 # INTEGRATION="true" while integration testing (false-y by default)
 ARG INTEGRATION
 ENV INTEGRATION=$INTEGRATION
-RUN gem install bundler -v '< 2'
 WORKDIR /usr/share/plugins/plugin
+RUN .ci/setup.sh
+RUN gem install bundler -v '< 2'
 RUN bundle install --with test ci
-COPY --chown=logstash:logstash . /usr/share/plugins/plugin
```

Comment:
please double check the copying - not sure why we had 3 `COPY`s instead of copy-ing everything in one step.

Reply:
The reason for the split COPY is that when you're developing locally and running the tests frequently, you don't want code changes to invalidate the docker layers that contain the computed `bundle install`, unless the changes touch the gemspec or Gemfile.

Reply:
your a saver, forgot about that and I actually experimented with the copy a while back and noticed the slow-down.
going to revert - simply wanted to be consistent to have all plugin code around before *setup.sh*.
maybe we want multiple hooks, eventually, given our current use of *setup.sh* it should be fine regardless of the plugin code being present or not.

Reply:
restored copy-ing `.` to exec after `bundle install` (as was the case previously), 
also the initial COPY now happens later + we need a new one due `.ci/setup.sh`